http/server.lua

```diff
@@ -589,7 +589,53 @@ response_mt = {
     }
 }
 
-local function handler(self, request)
+local function is_function(obj)
+    return type(obj) == 'function'
+end
+
+local function get_request_logger(server_opts, route_opts)
+    local root_logger = log.debug
```

Comment:
Simplification suggestion:
1) check route first, if found - return immediately, then
2) check server, if found - return immediately, then
3) return default

No need for `root_logger` variable.